README.md

```diff
@@ -43,17 +57,127 @@ Snippets for deployment template scaffolding and 70+ Azure resources.
+
+![Image showing the code lense for creating a parameter file association](./images/create-parameter-file.png)
+
+Select **New** from the context menu.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+Selecting **All parameters** adds every parameter from the template into the new parameter file. Selecting **Only required parameters** adds only parameters that do not have a default value specified in the template file.
+
+![Image showing options for adding parameters for parameter file (All parameters, only required parameters)](./images/create-parameter-file-3.png)
+
+Select a name and location for the parameter file.
+
+![Image showing a file save as dialog box](./images/create-parameter-file-4.png)
+
+Once done, a parameter file has been created. Where needed, remove the comment indicating that a parameter value is needed and add a parameter value.
+
+![Image showing new parameter file](./images/parameter-file.png)
+
+Also, notice that on the template file that the code lense text has been updated to reflect the current parameter file association.
+
+![Image showing that the parameter code lens has been updated to reflect associated parameter file](./images/code-lense.png)
+
+### Associate an existing parameter file
+
+To create an association with an existing parameter file, click on the code lense text `**Select or create a parameter file to enable full validation..**.
+
+![Image showing the code lense for creating a parameter file association](./images/create-parameter-file.png)
+
+Select **Browse** from the context menu and then select the parameter file.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+### Parameter validation
+
+Once a parameter file has been associated with a template, the extension provides parameter validation across both files. In the following example, a parameter is specified with a set of allowed values. In the parameter file, a value is specified that is not part of the allowed values list. Because of this, the extension produces an error.
+
+![Image showing a parameter validation issue](./images/param-file-validation.png)
+
+### Add missing parameters
+
+To add missing parameters to an existing parameter file, ensure that an association has been created, select `parameters`, and then click the light bulb icon.
+
+![Image showing the light bulb helper for adding missing parameters](./images/missing-params-one.png)
+
+Select the option to add only required or all missing parameters.
+
+![Image showing missing parameter options (Add missing required parameters, Add all missing parameters)](./images/missing-params-two.png)
 
-## Intellisense
+Enter a value for the added parameters and also remove the 'TODO' comment.
+
+![Image showing parameter file after missing parameters have been added](./images/missing-params-three.png)
+
```

Comment:
Since most of these work for parameters, variables, user namespaces and user functions, I'd suggest having a separate section for those, and then just point to those sections from here, rather than only mention them here in relation to parameters.

Reply:
Fixed